homeassistant/components/geo_location/usgs_earthquake_hazards_program_feed.py

```diff
@@ -0,0 +1,269 @@
+
+    def _generate_entity(self, external_id):
+        """Generate new entity."""
+        new_entity = UsgsEarthquakeHazardsProgramEvent(self, external_id)
+        # Add new entities to HA.
+        self._add_entities([new_entity], True)
+
+    def _update_entity(self, external_id):
+        """Update entity."""
+        dispatcher_send(self._hass, SIGNAL_UPDATE_ENTITY.format(external_id))
+
+    def _remove_entity(self, external_id):
+        """Remove entity."""
+        dispatcher_send(self._hass, SIGNAL_DELETE_ENTITY.format(external_id))
+
+
+class UsgsEarthquakeHazardsProgramEvent(GeoLocationEvent):
+    """This represents an external event with USGS Earthquake data."""
+
+    def __init__(self, feed_manager, external_id):
+        """Initialize entity with data from feed entry."""
+        self._feed_manager = feed_manager
+        self._external_id = external_id
+        self._name = None
+        self._distance = None
+        self._latitude = None
+        self._longitude = None
+        self._attribution = None
+        self._place = None
+        self._magnitude = None
+        self._time = None
+        self._updated = None
+        self._status = None
+        self._type = None
+        self._alert = None
+        self._remove_signal_delete = None
+        self._remove_signal_update = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self._remove_signal_delete = async_dispatcher_connect(
+            self.hass, SIGNAL_DELETE_ENTITY.format(self._external_id),
+            self._delete_callback)
+        self._remove_signal_update = async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_ENTITY.format(self._external_id),
+            self._update_callback)
+
+    @callback
+    def _delete_callback(self):
+        """Remove this entity."""
+        self._remove_signal_delete()
+        self._remove_signal_update()
+        self.hass.async_create_task(self.async_remove())
+
+    @callback
+    def _update_callback(self):
+        """Call update method."""
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def should_poll(self):
+        """No polling needed for USGS Earthquake events."""
+        return False
+
+    async def async_update(self):
+        """Update this entity from the data held in the feed manager."""
+        _LOGGER.debug("Updating %s", self._external_id)
+        feed_entry = self._feed_manager.get_entry(self._external_id)
+        if feed_entry:
+            self._update_from_feed(feed_entry)
+
+    def _update_from_feed(self, feed_entry):
+        """Update the internal state from the provided feed entry."""
+        self._name = feed_entry.title
+        self._distance = feed_entry.distance_to_home
+        self._latitude = feed_entry.coordinates[0]
+        self._longitude = feed_entry.coordinates[1]
+        self._attribution = feed_entry.attribution
+        self._place = feed_entry.place
```

Comment:
What does this tell us?

Reply:
Textual description of named geographic region near to the event.
(from https://github.com/home-assistant/home-assistant.io/blob/52e05314814bb654d89350bd4925f175f808d537/source/_components/geo_location.usgs_earthquake_hazards_program_feed.markdown)